### ISSUE TITLE
AnimationPlayer - Nodepath in a track in a animation stored in a PackedScene get their instance name automatically when saving the container scene.

### ISSUE THREAD

**Issue by Ranoller** [godotengine/godot#26144]
**Godot version:**

5e837b3f13ab1e3b31bb8d705e87820fa4eff21e

**OS/device including version:**

Win7

**Issue description:**

An Animation in PackedScene with an AnimationPlayer with a track called:
`EnemyPath2D/PathFollow2D:unit_offset`
is automatically renamed to:
`"../**InstancedPackedSceneName**/EnemyPath2D/PathFollow2D:unit_offset"`
on saving the Scene that contains that packed scene....

This change produces that **only the first packed scene** loaded in the **currentscene** can play the animation (because the **"InstancedPackedSceneName"** is stored in the animation), and only this instance can follow that track.  The packed scene remains inservible by itself (That packed scene **doesn´t have that track name outside the scene**). This behaviour happend with the unsaved resource or the animation stored in an independent tres file.

To note that InstancedPackedSceneName is unique per Node (godot doesn´t allow instances with the same name) and relative to every scene (you can change manually), so , **this name never should be added to the tracks from the original PackedScene** or the original resource (Godot should not change animation tracks names ever, this is not good behaviour, it can break to many things). If you change that track name manually, the animation is reproduced in all nodes, but on next saving, godot by itself destroy the reference other time.

**Minimal reproduction project:**

Will try...

**Comment by Ranoller** [comment]
Sorry about corrections, but this bug is annoying me and it was not clear the origin of that track name change...

**Comment by Ranoller** [comment]
Original packed scene with the animation:
![originalpackedscene](https://user-images.githubusercontent.com/11712737/53275161-4fcbf000-36fa-11e9-8402-0765b77762c0.png)

Working:
![originalworking](https://user-images.githubusercontent.com/11712737/53275172-578b9480-36fa-11e9-9a16-1ee69bfac3da.gif)

A Scene with 4 instances working:
![4instancesworking](https://user-images.githubusercontent.com/11712737/53275189-62462980-36fa-11e9-90d1-5c3168931c97.gif)

I add a new scene..... all works:

![5instancesworking](https://user-images.githubusercontent.com/11712737/53275206-6c682800-36fa-11e9-888d-aad343cb9330.gif)

I add other scene as a child of other.... all stills work:

![6instancesworking](https://user-images.githubusercontent.com/11712737/53275226-7e49cb00-36fa-11e9-9095-f31e9cef236f.gif)

I reparent and only one can follow the animation... the original track was renamed:

![reparentfailing](https://user-images.githubusercontent.com/11712737/53275309-db458100-36fa-11e9-8bcb-0e0c13401792.gif)

This is the auto-modified animation:

![modifiedoriginalanimation](https://user-images.githubusercontent.com/11712737/53275322-e6001600-36fa-11e9-8ff2-125fbbd2366e.png)

**Comment by Ranoller** [comment]
For the sake that you need to experiment by yourself:
[Godot3_1_bug_Catch.zip](https://github.com/godotengine/godot/files/2896104/Godot3_1_bug_Catch.zip)

**Comment by Ranoller** [comment]
I don´t know hot to do that in c++ and godot codebase, but in:
https://github.com/godotengine/godot/blob/master/editor/scene_tree_dock.cpp
In this lines:

```
1183	bool autorename_animation_tracks = bool(EDITOR_DEF("editors/animation/autorename_animation_tracks", true));
1184
1185	if (autorename_animation_tracks && Object::cast_to<AnimationPlayer>(p_base)) {
```

i think is needed to test if ... p_base is a PackedScene or something like that,

**Comment by Ranoller** [comment]
I think this issue it´s important to 3.1 stable, since can make a big headache to people without version control system....

**Comment by akien-mga** [comment]
The provided project seems to work fine in 3.2.2 RC 2 after reparenting the instance, can you confirm if it's solved for you too?

**Comment by Ranoller** [comment]
No, it´s not fixed. The problem is the "autorename" of nodes that godot performs. The operation that should be done to catch the bug is reparent a node, rename to some name that godot needs to change to not have duplicity(Ej. "Creature") , and reparent other time, and at this moment, godot changes the  name to "Creature6", the animation changes to ../ parent and the others nodes doesn´t perform the animation... If you follow the gifs, i think it´s clear the problem.

I tested in Godot 3.2.2 RC2 with the attached demo.

**Comment by Ranoller** [comment]
More specifically, the problem is that the "PackedScenes" shares the track names in the AnimationPlayer, and the Parent Scene of that instanced scene, changes all track names in the PackedScene instead in the animation track of the "Instanced Scene", breaking all other animations. But main problem described here is still valid.

I think in 2 solutions:
 - Doesn´t allow a Scene to change animation track names of the child instanced scenes by packed scenes in reparent. (Respecting encapsulation)
 - Store animation track changes in Parent scene instead change child instanced scenes (BTW deep change in dessign and complicated too much, breaking encapsulation and other "programmer things", but it´s posible, the same that you can change "modulate" and "transform" or "exported variables").

**Comment by Ranoller** [comment]
I want to note my comment of 23/02/2019 and the "autorename animation tracks" project setting. It´s still valid. The bug is related with this setting.

**Comment by KoBeWi** [comment]
Still valid in 3.3.2 and f030b0c

This is weird, because I remember similar issue being fixed. Maybe only some cases were fixed.
